python/pyspark/sql/tests.py

```diff
@@ -30,6 +30,19 @@
 import functools
 import time
 import datetime
+import traceback
+
+if sys.version_info[:2] <= (2, 6):
```

Comment:
this is different from how we check python version in other files, is this better than `sys.version >= '3'`?